riscv/triggers.cc

```diff
@@ -384,11 +384,13 @@ module_t::~module_t() {
 
 reg_t module_t::tdata1_read(unsigned index) const noexcept
 {
+  if (triggers.empty()) return 0;
```

Comment:
This doesn't seem necessary. If no triggers exist, should the `tdata1` CSR exist at all? The debug spec says about `tdata1` (and a bunch of other trigger-related CSRs):

> This register is optional if no triggers are implemented.

Reply:
I have a very slight preference towards providing the CSRs in the optional case, since the existing riscv-tests breakpoint tests assumes so. Those should still be fixed though.

I think these checks should be preserved either way, however, since a future libriscv user or spike fork may choose to provide the CSRs in a legal implementation.

Reply:
How about instantiating all those CSRs as `const_csr_t` when no triggers exist? In `state_t::reset()`.